### DISCUSSION TITLE
MIDI support on FreeBSD

### DISCUSSION THREAD

**Discussion by watchmakerful** [audacious-media-player/audacious#1493] [Q&A] [answered]
Does Audacious support MIDI on FreeBSD? I don't see any mention of the MIDI plugin in the settings (other plugins are present and working). What else should I install?

**Comment by radioactiveman** [comment]
@madpilot78: Could you answer these questions please? Is the "amidiplug" plugin for Audacious not available on FreeBSD?

**Comment by madpilot78** [comment, marked as answer]
@watchmakerful How are you installing audacious? Via binary packages?

@radioactiveman, @watchmakerful  the audacious port in the FreeBSD ports tree does provide an option to compile Amidi plugin but that option is disabled by default, so the official binary package (which is compiled with default option) does not provide this plugin.

@watchmakerful can compile it himself enabling the option (how to do this is documented in the [handook ports chapter](https://docs.freebsd.org/en/books/handbook/ports/#ports-using).

Default options are chosen to try to strike  a middle ground between features and reducing dependencies. But I started maintaining the port when it was already created, and did not choose the default options myself, I kept them as they are, so I can't really tell much about why amidi support was deemed unnecessary.

@watchmakerful if you'd like to ask for default options to be modified please file a request at https://bugs.freebsd.org for the `multimedia/audacious-plugins` port stating the request and I'll see if it can be done without pulling in a major number of dependencies. Usually FreeBSD users prefer lean binary packages with not too many dependencies so we are cautious with adding them.

**Reply by watchmakerful** [reply]
>>How are you installing audacious? Via binary packages?

Yes, via pkg.

>>if you'd like to ask for default options to be modified please file a request at https://bugs.freebsd.org/ for the multimedia/audacious-plugins port

I'll try it.

**Comment by watchmakerful** [comment]
Huge thanks, the latest pkg now has this plugin enabled by default.
